### PR TITLE
XZ compression

### PR DESCRIPTION
**Pull request by phunkyfish**
v7.6.0
- Added: Support url-tvg in M3U header in addition to x-tvg-url
- Added: Add support for XZ compression of XMLTV data
- Fixed: Allow flusonnic channel ids containing forward slashes
- Fixed: Disable other catchup settings if catched is not enabled
- Added: Add setting for catchup correction
- Added: Support xeev XC catchup prefixes

### CODE REVIEW

**Comment by phunkyfish** [comment]
@AlwinEsch I added XZ decompression to the add-on using depends. But for some reason it doesn't want to build on windows or android and I'm not sure why. Any ideas?

**Comment by phunkyfish** [comment]
> @AlwinEsch I added XZ decompression to the add-on using depends. But for some reason it doesn't want to build on windows or android and I'm not sure why. Any ideas?

Figured it out eventually. Android was a typo and a special define was needed to Windows ;)

File: Findlzma.cmake [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# - Try to find pugixml
```
Comment (fuzzard) [inline Findlzma.cmake]: Copypasta

File: pvr.iptvsimple/changelog.txt [hunk 2]
```diff
@@ -1,3 +1,7 @@
+v7.6.0
+- Added: Support url-tvg in M3U header in adion to x-tvg-url
```
Comment (fuzzard) [inline pvr.iptvsimple/changelog.txt]: Typo addition

File: src/iptvsimple/utilities/FileUtils.cpp [hunk 3]
```diff
@@ -132,6 +133,44 @@ bool FileUtils::GzipInflate(const std::string& compressedBytes, std::string& unc
...
+  if (compressedBytes.size() == 0)
+  {
+    uncompressedBytes = compressedBytes;
+    return true;
+  }
+
+  uncompressedBytes.clear();
+
+  lzma_stream strm = LZMA_STREAM_INIT;
+	lzma_ret ret = lzma_stream_decoder(
```
Comment (fuzzard) [inline src/iptvsimple/utilities/FileUtils.cpp]: Tab/space issue here and the next few lines?

Comment (phunkyfish) [reply src/iptvsimple/utilities/FileUtils.cpp]: Done

File: pvr.iptvsimple/changelog.txt (hunk 2 above)
Comment (phunkyfish) [reply pvr.iptvsimple/changelog.txt]: Done

File: Findlzma.cmake (hunk 1 above)
Comment (phunkyfish) [reply Findlzma.cmake]: Done

File: src/iptvsimple/utilities/FileUtils.cpp [hunk 4]
```diff
@@ -132,6 +133,43 @@ bool FileUtils::GzipInflate(const std::string& compressedBytes, std::string& unc
...
+    uncompressedBytes = compressedBytes;
+    return true;
+  }
+
+  uncompressedBytes.clear();
+
+  lzma_stream strm = LZMA_STREAM_INIT;
+  lzma_ret ret = lzma_stream_decoder(&strm, UINT64_MAX, LZMA_TELL_UNSUPPORTED_CHECK | LZMA_CONCATENATED);
+
+	if (ret != LZMA_OK)
```
Comment (fuzzard) [inline src/iptvsimple/utilities/FileUtils.cpp]: Missed one, haha

Comment (phunkyfish) [reply src/iptvsimple/utilities/FileUtils.cpp]: Done

**Pull request merged**